### PR TITLE
No backward in the feature extraction network

### PR DESCRIPTION
**Pull request by kloudkl**
Because the learning rate is default to be 1 and backward is default to be true (#389), even though the phase is set to `Caffe::TEST`, the feature extraction network still performs backward propagation unnecessarily.

### CODE REVIEW

**Comment by Noiredd** [comment]
If no backward propagation is not explicitly called, are the gradients still being calculated during inference? I'm optimizing a network for inference speed, and for this particular application I'm not interested in backprop - `caffe time` still includes it in measurements though. Ideally, I'd like to be able to train a network and then disable all unnecessary computations in order to benchmark it. Do I have to modify the entire proto, or is there some known best way to approach this?
